djautotask/api.py

```diff
@@ -588,6 +588,19 @@ def delete(self, instance, parent=None):
         # AT sends deleted_id or 500 in the case failure instead of 404 or 204
         return response.get('itemId')
 
+    def _set_impersonation_id(self, impersonation_id, target=None):
+        from djautotask.models import Resource
+        checked_impersonation_id = None
+        if impersonation_id:
+            try:
+                resource = Resource.objects.get(id=impersonation_id)
```

Comment:
I said you should just pass the resource instance directly to the update method. You already have it everywhere the update method is called, why just pass in the resource ID if you are just going to query for it again?

Can you explain how a mid-file import in a function to query for an instance you could have passed in is better than just passing the instance in?

Reply:
I thought `**key ` can be still used without change.

As you said, it sends additional query. I will fix that, too.

Reply:
Ok, I explained that I have a problem with that whole `**key` method of passing in just the ID already. It's better to just pass in the resource in cases where applicable. That way there are two cases from TopLeft's perspective: 
* Impersonation is requested -> pass resource as an argument
* Impersonation is NOT requested -> don't pass the resource

Then all the logic if it CAN use impersonation exists in `django-autotask`, rather than TopLeft having all sorts of complicated hacks to check if it can. And if `django-autotask` doesn't receive a resource it just makes the request normally.

So, please TopLeft and `django-autotask` to pass and accept the resource instance.

Reply:
Done now~. Sorry for this extra review and fix. :sweat: